spec/OpenLineage.yml

```diff
@@ -0,0 +1,250 @@
+openapi: 3.0.2
+info:
+  title: OpenLineage
+  version: 0.0.1
+  description: OpenLineage is an open source **lineage and metadata collection API** for the data ecosystem.
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+paths:
+  /events:
+    post:
+      summary: Updates the state of a run
+      description: Updates a run state for a job.
+      operationId: postRunStateUpdate
+      tags:
+        - OpenLineage
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RunStateUpdates'
+      responses:
+        '200':
+          description: OK
+
+components:
+  schemas:
+    RunStateUpdates:
+      type: array
+      items:
+        $ref: '#/components/schemas/RunStateUpdate'
+
+    RunStateUpdate:
+      type: object
+      properties:
+        runId:
+          description: the unique identifier for that run
+          type: string
+          format: uuid
+        jobId:
+          $ref: '#/components/schemas/JobId'
```

Comment:
The jobid can be rolled into the job vertex and marked as required to achieve the same effect, since all the other job attributes are optional. If optional job fields are present at write, it can be treated as an upsert to a standalone job entity.

Reply:
I agree with this statement. I moved the jobId here because the runId can be linked to only one jobId. They are both  required definitions for the RunStateUpdate and they must be constant for all StateUpdates of the same run. Nesting JobId in `job` gives the false impression that more than one job can be linked to the run.
I'm open to other suggestions as well.

Reply:
This also simplified the notion of parent run and job. 
I would consider the runid to be the composite (jobId, runID)